### PR TITLE
Unpin animation

### PR DESCRIPTION
**Pull request by apalaniuk**
Adds an animation on pinning/unpinning course tiles.

Also refactors common behavior of My Courses/All Courses into Polymer behavior, so they can handle course entity management independently; d2l-all-courses was previously depending on d2l-my-courses for state management.

### CODE REVIEW

File: d2l-course-management-behavior.html [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+			ready: function() {
+				this._announceElement = document.createElement('d2l-offscreen');
+				this._announceElement.setAttribute('aria-live', 'polite');
+			},
+			setPinnedCoursesFromSirenResponse: function(response) {
+				var parser = document.createElement('d2l-siren-parser');
+				var allEnrollments = parser.parse(response);
+				var allCourses = allEnrollments.getSubEntitiesByClass('course-offering');
+
+				this.set('pinnedCoursesEntities', allCourses.filter(function(item) {
```
Comment (ryantmer) [inline d2l-course-management-behavior.html]: The effect is the same, but you can use the siren parser again here - `this.pinnedCoursesEntities = allCourses.getSubEntitiesByClass('active')`.

Comment (ryantmer) [inline d2l-course-management-behavior.html]: Wait, nvm. Forgot that returns an array, which you can't `getSubEntitiesByClass` on. Carry on!

Comment (ryantmer) [inline d2l-course-management-behavior.html]: Ah, I remember why I was confused. At one point there was discussion of having `getSubEntitiesByClass(['course-offering', 'active'])` be a thing. Never happened though. Also might have been an `or` on the class names, not an `and`. So, doubly useless in this context.

Sorry for comment spam. :meat_on_bone:

File: d2l-course-management-behavior.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+					return item.hasClass('active');
+				}));
+
+				this._hasPinnedCourses = this.pinnedCoursesEntities.length > 0;
+				this._hasCourses = this._hasPinnedCourses;
+			},
+			setAllCoursesFromSirenResponse: function(response) {
```
Comment (ryantmer) [inline d2l-course-management-behavior.html]: Hmm... now that I'm looking at these side-by-side, can we just ditch one of these response handlers? The only difference is that the one for pinned has one less level of nesting, but due to the circumstances it's not actually going to save anything anyway :balloon:

File: d2l-course-tile.html [hunk 3]
```diff
@@ -103,9 +109,20 @@
 			behaviors: [
 				Polymer.D2L.MyCourses.LocalizeBehavior
 			],
+			listeners: {
+				'touch-pin-hover': '_onUnpinHover',
+				'touch-pin-select': 'pinClickHandler',
+				'animationend': '_onAnimationComplete'
```
Comment (ryantmer) [inline d2l-course-tile.html]: (assuming this isn't a neon thing), can this be `animation-end` for consistency? (although I assume this is a neon thing)

Comment (apalaniuk) [inline d2l-course-tile.html]: It's not a neon animation - it's the [CSS animationend event](https://developer.mozilla.org/en-US/docs/Web/Events/animationend), so I'm not sure if there's a good way to align naming?

Comment (ryantmer) [inline d2l-course-tile.html]: Ah! Nvm then, all good as-is :)

**Comment by ryantmer** [comment]
Many :+1:s!

**Pull request merged**